### ISSUE TITLE
Connect to MQTT 5 broker

### ISSUE THREAD

**Issue by nosovk** [cotonic/cotonic#43]
I try to use cotonic as MQTT client instead of mqtt.js
But currently I can't find how to connect to EMQ X broker.

There is a test EMQ X broker available https://www.emqx.io/mqtt/public-mqtt5-broker

I try to connect to it using

```javascript
cotonic.mqtt_bridge.newBridge("broker.emqx.io:8083",
                              {protocol: "ws", controller_path:"/mqtt"});
```

It throws error
`Error during WebSocket handshake: Unexpected response code: 400`
According to doc: https://docs.emqx.io/en/broker/latest/advanced/http-api.html#response-code
400 - Invalid client request, such as wrong request body or parameters

I've tested broker using http://tools.emqx.io/ and it works, it means that problem is in snippet above, but can't understand what's wrong with that conntection.

**Comment by mworrell** [comment]
Guess EMQ doesn't like something we are sending along.

Will experiment a bit and might and check their code.

**Comment by mworrell** [comment]
It turns out to be an error in the EMQ code.

EMQ accepts in the `Sec-Websocket-Protocol` a single protocol, where in fact it should be able to handle multiple protocols.

Besides that it also accepts anything starting with `mqtt`, and tells the remote that it is supporting that protocol. This is an error, as the EMQ web socket handler only handles the `mqtt` protocol and not any extension of it.

```erlang
    case cowboy_req:parse_header(<<"sec-websocket-protocol">>, Req) of
        undefined ->
            %% TODO: why not reply 500???
            {cowboy_websocket, Req, [Req, Opts], WsOpts};
        [<<"mqtt", Vsn/binary>>] ->
            Resp = cowboy_req:set_resp_header(
                     <<"sec-websocket-protocol">>, <<"mqtt", Vsn/binary>>, Req),
            {cowboy_websocket, Resp, [Req, Opts], WsOpts};
        _ ->
            {ok, cowboy_req:reply(400, Req), WsOpts}
    end.
```

I will add an issue for EMQ.

In the meantime, if you change the following line in `src/cotonic.mqtt_transport.ws.js`:

```javascript
self.socket = new WebSocket( self.remoteUrl, [ "mqtt.cotonic.org", "mqtt" ] );
```

to:

```javascript
// EMQ is erronously accepting any protocol starting with `mqtt`, so it accepts
// 'mqtt.cotonic.org', so that we start the handshake.
// self.socket = new WebSocket( self.remoteUrl, [ "mqtt.cotonic.org", "mqtt" ] );
self.socket = new WebSocket( self.remoteUrl, [ "mqtt" ] );
```

Then it will connect.

**Comment by mworrell** [comment]
I have pushed the change above to Cotonic master.

As this kind of errors might happen more often with brokers we might want to introduce another method for initiating the handshake.

**Comment by mworrell** [comment]
Issue in EMQ https://github.com/emqx/emqx/issues/3914

**Comment by mworrell** [comment]
This issue has been fixed in EMQ v4.3-alpha.1
